tensorflow/core/grappler/optimizers/auto_parallel.cc

```diff
@@ -166,7 +170,17 @@ Status AutoParallel::Initialize(const GrapplerItem& item) {
   std::set<string> dont_replicate_nodes;
   for (const auto& variable : item.MainVariables()) {
     dont_replicate_nodes.insert(variable->name());
+    VariableDef var = variables_[strings::StrCat(variable->name(), ":0")];
+    if (!var.initializer_name().empty()) {
+      dont_replicate_nodes.insert(
+        var.initializer_name().substr(0, var.initializer_name().size() - 2));
```

Comment:
Could you use this function here? https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/utils.cc#L109